### DISCUSSION TITLE
Support for AMD and ARM

### DISCUSSION THREAD

**Discussion by jcisio** [sustainable-computing-io/kepler#1290] [General]
Hello,
We are looking at the current Kepler status, checking what is currently supported. Unless we missed something, looking at "sensors" code, the support is mostly for Intel platform. For example, RAPL-msr has full support for Intel (package/DRAM/PP0/PP1) and no support at all for either AMD or ARM. Is it correct or there is another mechanism somewhere?
Thank you.

**Comment by rootfs** [comment]
Ampere ARM is supported https://github.com/sustainable-computing-io/kepler/blob/main/pkg/sensors/components/source/apm_xgene_sysfs.go

AMD should work but I haven't got a setup to verify.

**Reply by ErwanAliasr1** [reply]
@rootfs I've been reading the rapl code of the project (https://github.com/sustainable-computing-io/kepler/blob/main/pkg/sensors/components/source/rapl_msr_util.go#L38) and the MSR shown here and very intel specific.

On EPYC, the per_core power consumption has to be read via MSR 0xc001029a and the package from 0xc001029b.
To get other power metrics from the EPYC socket, we'd have also to use the HSMP interface (which can be reached from recent kernel cf https://docs.kernel.org/arch/x86/amd_hsmp.html)

**Reply by rootfs** [reply]
thank you @ErwanAliasr1 for the investigation. Would you please get a PR for EPYC?
